<?xml version="1.0"?>
<launch>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find capsicum_superellipsoid_detector)/launch/rviz.rviz"/>

<!-- COMMENTED OUT TEMPORARLY
  <node pkg="capsicum_superellipsoid_detector" type="detector_node" name="capsicum_superellipsoid_detector" output="screen" respawn="true">
    <remap from="pc_in" to="/camera/depth/points"/>
  </node>
-->

  <node pkg="nodelet" type="nodelet" name="depth_proc_point_cloud_xyzrgb" args="standalone depth_image_proc/point_cloud_xyzrgb" output="screen">
    <param name="queue_size" value="300"/>
    <param name="exact_sync" value="false"/>
    <remap from="rgb/camera_info" to="/camera/color/camera_info"/>
    <remap from="rgb/image_rect_color" to="/camera/color/image_raw"/>
    <remap from="depth_registered/image_rect" to="/roi_depth"/>
    <!-- <remap from="depth_registered/image_rect" to="/camera/aligned_depth_to_color/image_raw"/> -->
  </node>


  <node pkg="capsicum_superellipsoid_detector" type="mask_filter.py" name="capsicum_superellipsoid_detector" output="screen">
    <remap from="input_depth" to="/camera/aligned_depth_to_color/image_raw"/>
    <remap from="input_detection" to="/agrobot_mrcnn_ros_node/detections"/>
  </node>

  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server" screen="output">
    <param name="resolution" value="0.002" />
    <param name="frame_id" type="string" value="world" />
    <param name="sensor_model/hit" value="0.55" />
    <param name="sensor_model/max_range" value="3.0" />
    <remap from="cloud_in" to="/depth_registered/points" />
  </node>

</launch>
